### PR TITLE
Add default for `wp_is_development_mode()`

### PR DESCRIPTION
**Pull request by joemcgill**
This allows `wp_is_default_mode()` to be called with a default value of 'any' in order to return true if any valid default mode is set. This was requested in [#55](https://core.trac.wordpress.org/ticket/57487#comment:55).

Note that this includes changes from `e5fe64b0` which will be handled in https://github.com/WordPress/wordpress-develop/pull/4850.

Trac ticket: https://core.trac.wordpress.org/ticket/58855

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by felixarntz — CHANGES_REQUESTED** [review]
See https://core.trac.wordpress.org/ticket/57487#comment:62 why I think that (at least at this point) this is not a good idea.

File: src/wp-includes/load.php [hunk 1]
```diff
@@ -322,11 +322,13 @@ function wp_get_development_mode() {
  *
  * @since 6.3.0
  *
- * @param string $mode Development mode to check for. Either 'core', 'plugin', 'theme', or 'all'.
+ * @param string $mode Optional. Development mode to check for. Either 'core', 'plugin',
+ *                     'theme', all', or 'any'. Defaults to 'any'.
  * @return bool True if the given mode is covered by the current development mode, false otherwise.
  */
-function wp_is_development_mode( $mode ) {
+function wp_is_development_mode( $mode = 'any' ) {
```
Comment (azaozz) [inline src/wp-includes/load.php:329]:
```suggestion
function wp_is_development_mode( $mode = null ) {
```

See below.

File: src/wp-includes/load.php [hunk 2]
```diff
@@ -336,6 +338,18 @@ function wp_is_development_mode( $mode ) {
...
 
+	$valid_modes = array(
+		'core',
+		'plugin',
+		'theme',
+		'all',
+	);
+
+	// Return true if checking for 'any' mode and a valid one is set.
+	if ( 'any' === $mode && in_array( $current_mode, $valid_modes, true ) ) {
```
Comment (azaozz) [inline src/wp-includes/load.php:349]:
```suggestion
	if ( null === $mode && in_array( $current_mode, $valid_modes, true ) ) {
```

File: src/wp-includes/load.php [hunk 3]
```diff
@@ -336,6 +338,18 @@ function wp_is_development_mode( $mode ) {
 		return true;
 	}
 
+	$valid_modes = array(
+		'core',
+		'plugin',
+		'theme',
+		'all',
+	);
```
Comment (azaozz) [inline src/wp-includes/load.php:346]:
```suggestion
	$valid_modes = array(
		'core',
		'plugin',
		'theme',
	);
```

**Review by azaozz — COMMENTED** [review]
Think that your suggestion here https://core.trac.wordpress.org/ticket/57487#comment:58 is best.

**Pull request closed without merging**